Baystef/Quick-Credit
# #165738896 Build out endpoints for GET All Loans, Repaid Loans and Unrepaid Loans

#### What does this PR do?
Have the GET all loan endpoint and the repaid and unrepaid loan query endpoints working

#### Description of Task to be completed?
Have the following endpoints working:
`GET /api/v1/loans/`
`GET /api/v1/loans/?status=approved&repaid=false`
`GET /api/v1/loans/?status=approved&repaid=true`

#### How should this be manually tested?
After cloning the repo,
* `cd quick-credit`
* Run `npm run start` on the console while in the quick-credit directory
* On Postman, make a HTTP POST to `http://localhost:4700/api/v1/auth/signin/`
with request body 
```javascript
{
    "email": "admin@quickcredit.com",
    "password": "quickcreditsecret"
}
```
The response body should be
```javascript
{
    "status": 200,
    "data": {
        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7ImlkIjoxLCJlbWFpbCI6ImFkbWluQHF1aWNrY3JlZGl0LmNvbSIsImlzQWRtaW4iOnRydWV9LCJpYXQiOjE1NTczMDM5OTMsImV4cCI6MTU1NzkwODc5M30.hN1iFtEmpzts5-sQU7zZQForpVzXmvldhSvqCVP_Uk4",
        "id": 1,
        "firstName": "Admin",
        "lastName": "Quickcredit",
        "email": "admin@quickcredit.com",
        "password": "$2b$08$cxMMrlkCnCktPBEBwr9VtuHfY35GzxCYH6J6RjqHQPvbWpyEcvMa2",
        "phoneNo": 2347032326464,
        "homeAddress": "1, Quick Credit Avenue",
        "workAddress": "2, Quick Credit Complex",
        "status": "verified",
        "isAdmin": true,
        "createdAt": "5/8/2019, 9:26:33 AM"
    }
}
```
* Copy the token value at `{ data.token }` from the response
* On Postman, goto **Headers** tab and then set **New Key** to `Authorization` and  **Value** to 
` Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7ImlkIjoxLCJlbWFpbCI6ImFkbWluQHF1aWNrY3JlZGl0LmNvbSIsImlzQWRtaW4iOnRydWV9LCJpYXQiOjE1NTczMDM5OTMsImV4cCI6MTU1NzkwODc5M30.hN1iFtEmpzts5-sQU7zZQForpVzXmvldhSvqCVP_Uk4` i.e `Bearer { data.token }`

* Now goto `http://localhost:4700/api/v1/loans/` and make a GET request 
The response should be
```javascript
{
    "status": 200,
    "data": [
        {
            "id": 1,
            "user": "daramola.steve@gmail.com",
            "loanAmount": 50000,
            "tenor": 5,
            "status": "approved",
            "repaid": false,
            "interest": 2500,
            "paymentInstallment": 10500,
            "balance": 52500,
            "createdAt": "5/6/2019, 1:37:13 PM"
        },
        {
            "id": 2,
            "user": "daramola@quic.com",
            "loanAmount": 100000,
            "tenor": 7,
            "status": "pending",
            "repaid": false,
            "interest": 5000,
            "paymentInstallment": 15000,
            "balance": 105000,
            "createdAt": "5/6/2019, 1:42:42 PM"
        },
...... ]
}
```

* Goto `http://localhost:4700/api/v1/loans/?status=approved&repaid=false` and make a GET request 
The response should be
```javascript
{
    "status": 200,
    "data": [
        {
            "id": 1,
            "user": "daramola.steve@gmail.com",
            "loanAmount": 50000,
            "tenor": 5,
            "status": "approved",
            "repaid": false,
            "interest": 2500,
            "paymentInstallment": 10500,
            "balance": 52500,
            "createdAt": "5/6/2019, 1:37:13 PM"
        }
    ]
}
```
* Goto `http://localhost:4700/api/v1/loans/?status=approved&repaid=true` and make a GET request 
The response should be
```javascript
{
    "status": 200,
    "data": [
        {
            "id": 4,
            "user": "daramola@yahoo.com",
            "loanAmount": 100000,
            "tenor": 7,
            "status": "approved",
            "repaid": true,
            "interest": 5000,
            "paymentInstallment": 15000,
            "balance": 105000,
            "createdAt": "5/6/2019, 1:42:42 PM"
        },
        {
            "id": 5,
            "user": "daramola@hotmail.com",
            "loanAmount": 100000,
            "tenor": 7,
            "status": "approved",
            "repaid": true,
            "interest": 5000,
            "paymentInstallment": 15000,
            "balance": 105000,
            "createdAt": "5/6/2019, 1:42:42 PM"
        }
    ]
}
```

#### Any background context you want to provide?
GET All Loans, Repaid Loans and Unrepaid Loans can only be accessed by the admin
which is why you have to feed the admin token into the headers before making a request

#### What are the relevant pivotal tracker stories?
[#165738896](https://www.pivotaltracker.com/story/show/165738896)
[#165738956](https://www.pivotaltracker.com/story/show/165738956)
[#165738977](https://www.pivotaltracker.com/story/show/165738977)

#### Questions:
Nil
